pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateTableViewSyncer.java

```diff
@@ -0,0 +1,282 @@
+ * under the License.
+ */
+
+package org.apache.pulsar.broker.loadbalance.extensions.channel;
+
+import static org.apache.pulsar.broker.ServiceConfiguration.ServiceUnitTableViewSyncerType.SystemTopicToMetadataStoreSyncer;
+import static org.apache.pulsar.broker.loadbalance.extensions.ExtensibleLoadManagerImpl.COMPACTION_THRESHOLD;
+import static org.apache.pulsar.broker.loadbalance.extensions.ExtensibleLoadManagerImpl.configureSystemTopics;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pulsar.broker.PulsarService;
+import org.apache.pulsar.common.util.FutureUtil;
+import org.apache.pulsar.common.util.ObjectMapperFactory;
+
+/**
+ * Defines ServiceUnitTableViewSyncer.
+ * It syncs service unit(bundle) states between metadata store and system topic table views.
+ * One could enable this syncer before migration from one to the other and disable it after the migration finishes.
+ */
+@Slf4j
+public class ServiceUnitStateTableViewSyncer implements Closeable {
+    private static final int MAX_CONCURRENT_SYNC_COUNT = 100;
+    private static final int SYNC_WAIT_TIME_IN_SECS = 300;
+    private PulsarService pulsar;
+    private volatile ServiceUnitStateTableView systemTopicTableView;
+    private volatile ServiceUnitStateTableView metadataStoreTableView;
+    private volatile boolean isActive = false;
+
+
+    public void start(PulsarService pulsar)
+            throws IOException, TimeoutException, InterruptedException, ExecutionException {
+        if (!pulsar.getConfiguration().isLoadBalancerServiceUnitTableViewSyncerEnabled()) {
+            return;
+        }
+
+        if (isActive) {
+            return;
+        }
+        this.pulsar = pulsar;
+
+        try {
+
+            syncExistingItems();
+            // disable compaction
+            if (!configureSystemTopics(pulsar, 0)) {
+                throw new IllegalStateException("Failed to disable compaction");
+            }
+            syncTailItems();
+
+            isActive = true;
+
+        } catch (Throwable e) {
+            log.error("Failed to start ServiceUnitStateTableViewSyncer", e);
+            throw e;
+        }
+    }
+
+    private CompletableFuture<Void> syncToSystemTopic(String key, ServiceUnitStateData data) {
+        return systemTopicTableView.put(key, data);
+    }
+
+    private CompletableFuture<Void> syncToMetadataStore(String key, ServiceUnitStateData data) {
+        return metadataStoreTableView.put(key, data);
+    }
+
+    private CompletableFuture<Void> dummy(String key, ServiceUnitStateData data) {
```

Comment:
Why return a `CompletableFuture<Void>` here? I see the return value has never been used.

Reply:
This returns future in case it needs to be chained in async.

Reply:
updated it to return void.